builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -675,34 +678,45 @@ func resourceAwsSecurityGroupUpdateRules(
 	return nil
 }
 
+func updateSecurityGroupCache(conn *ec2.EC2, cache *Cache) error {
+	mtKey := "security_groups_cache"
+	awsMutexKV.Lock(mtKey)
+	if cache.SecurityGroups == nil {
+		req := &ec2.DescribeSecurityGroupsInput{}
+		resp, err := conn.DescribeSecurityGroups(req)
+
+		if err != nil {
+			log.Printf("Error on SGStateRefresh: %s", err)
+			return err
+		}
+
+		cache.SecurityGroups = resp
+	}
+	awsMutexKV.Unlock(mtKey)
```

Comment:
Also, it looks like we only really update the group cache if the existing on is nil. So if a new one is created, some operations happen, and we need to then consult this cache, the new SG won't be in it, right?